proxstar/static/js/script.js

```diff
@@ -891,3 +891,57 @@ $(document).on('focus click', "[id^=boot-order-]", function() {
         }
     });
 });
+
+$(".rename-vm").click(function(){
+    const vmid = $(this).data('vmid');
+    const old_name = $(this).data('old_name');
+    swal({
+        title: 'Enter what you would like this VM to be renamed to:',
+        content: {
+            element: 'input',
+            attributes: {
+                type: 'string',
+            },
+        },
+        buttons: {
+            cancel: {
+                text: "Cancel",
+                visible: true,
+                closeModal: true,
+                className: "",
+            },
+            confirm: {
+                text: "Select",
+                closeModal: false,
+            }
+        },
+    })
+    .then((new_name) => {
+        if (new_name) {
+            fetch(`/starrs/${vmid}/hostname/${old_name}/${new_name}`, {
```

Comment:
Probably better to do something like `/vm/${vmid}/rename` and include the new name in the POST data